marshal/error.go

```diff
@@ -0,0 +1,7 @@
+package marshal
+
+import "errors"
+
+var ErrorGOCQL = errors.New("gocql error")
+var ErrorMarshal = errors.Join(ErrorGOCQL, errors.New("marshal error"))
+var ErrorUnmarshal = errors.Join(ErrorGOCQL, errors.New("unmarshal error"))
```

Comment:
Everything that comes from `gocql` package is a `gocql error`, so adding `gocql error` to any error you produce it does not make sense, if use wants they should add it on their own.
```suggestion
var ErrMarshal = errors.New("marshal error")
var ErrUnmarshal = errors.New("unmarshal error")
```

Reply:
My research on the problem with "errors" shows the following:
1) Each error has its own `source` and the `path` it follows, and it is not the same thing.
2) In cases where the error is caused by limitations other packages, the users of the package are most interested not in the `path`, but in the `source` of the error.

Therefore, it is logical to give users the opportunity to easily identify the `source` of the error. 
This can be done by entering `ones error` for the entire package.
And using `ones error` only for cases where the error is caused by limitations in this package and in the used simple packages.
For all other cases: either throwing as is, or splitting by the source (e.g. `ErrorNet`,`ErrorDB`).

If you don't agree with this, let's make your version.

Reply:
Please write me what you have chosen.

Reply:
> My research on the problem with "errors" shows the following:
> 
> 1. Each error has its own `source` and the `path` it follows, and it is not the same thing.
> 2. In cases where the error is caused by limitations other packages, the users of the package are most interested not in the `path`, but in the `source` of the error.
> 
> Therefore, it is logical to give users the opportunity to easily identify the `source` of the error. This can be done by entering `ones error` for the entire package. And using `ones error` only for cases where the error is caused by limitations in this package and in the used simple packages. For all other cases: either throwing as is, or splitting by the source (e.g. `ErrorNet`,`ErrorDB`).
> 
> If you don't agree with this, let's make your version.

They can easily identify source on their own by joining, or wrapping error they going to get from driver API.

Reply:
Done